Samples/VirtualCamera/VirtualCamera_Installer/main.cpp

```diff
@@ -0,0 +1,335 @@
+        RETURN_IF_FAILED(spSourceReader->SetStreamSelection(streamIdx, TRUE));
+        RETURN_IF_FAILED(spSourceReader->SetCurrentMediaType(streamIdx, NULL, spMediaType.get()));
+        RETURN_IF_FAILED_MSG(MediaSourceUT_Common::ValidateStreaming(spSourceReader.get(), streamIdx, spMediaType.get()), "Streaming validation failed");
+        RETURN_IF_FAILED(spSourceReader->SetStreamSelection(streamIdx, FALSE));
+        LOG_COMMENT(L"Streaming validation passed!");
+    }
+    return S_OK;
+}
+
+//
+// CONSOLE
+//
+winrt::hstring SelectVirtualCamera()
+{
+    winrt::hstring strSymLink;
+
+    std::vector<DeviceInformation> vcamList;
+    if (FAILED(VCamUtils::GetVirtualCamera(vcamList)) || (vcamList.size() == 0))
+    {
+        LOG_COMMENT(L"No Virtual Camera found ");
+        return strSymLink;
+    }
+
+    for (uint32_t i = 0; i < vcamList.size(); i++)
+    {
+        auto dev = vcamList[i];
+        LOG_COMMENT(L"[%d] %s (%s) ", i + 1, dev.Id().data(), dev.Name().data());
+    }
+    LOG_COMMENT(L"select device ");
+    uint32_t devIdx = 0;
+    std::wcin >> devIdx;
+
+    if (devIdx <= 0 || devIdx > vcamList.size())
+    {
+        LOG_COMMENT(L"Invalid device selection ");
+        return strSymLink;
+    }
+
+    strSymLink = vcamList[devIdx - 1].Id();
+    return strSymLink;
+}
+
+DeviceInformation SelectPhysicalCamera()
+{
+    winrt::hstring strSymLink;
+    DeviceInformation devInfo{ nullptr };
+
+    std::vector<DeviceInformation> camList;
+    if (FAILED(VCamUtils::GetPhysicalCameras(camList)) || (camList.size() == 0))
+    {
+        LOG_COMMENT(L"No physical Camera found");
+        return devInfo;
+    }
+
+    for (uint32_t i = 0; i < camList.size(); i++)
+    {
+        auto dev = camList[i];
+        LOG_COMMENT(L"[%d] %s \n", i + 1, dev.Id().data());
+    }
+    LOG_COMMENT(L"select device");
+    uint32_t devIdx = 0;
+    std::wcin >> devIdx;
+
+    if (devIdx <= 0 || devIdx > camList.size())
+    {
+        LOG_COMMENT(L"Invalid device selection");
+        return devInfo;
+    }
+
+    return camList[devIdx - 1];
+}
+
+HRESULT SelectRegisterVirtualCamera(_Outptr_ IMFVirtualCamera** ppVirtualCamera)
+{
+    while (true)
+    {
+        LOG_COMMENT(L"\n select opitions: \n 1 - VCam-SimpleMediaSource \n 2 - VCam-HWMediaSource \n 3 - quit \n");
+        uint32_t select = 0;
+        std::wcin >> select;
+
+        wil::com_ptr_nothrow<IMFVirtualCamera> spVirtualCamera;
+        switch (select)
+        {
+        case 1:
+        {
+            SimpleMediaSourceUT test;
+            RETURN_IF_FAILED(test.CreateVirtualCamera(ppVirtualCamera));
+            break;
+        }
+        case 2: 
+        {
+            auto devInfo = SelectPhysicalCamera();
+            
+            HWMediaSourceUT test(devInfo.Id());
+            RETURN_IF_FAILED(test.CreateVirtualCamera(devInfo.Name(), ppVirtualCamera));
+            break;
+        }
+
+        default:
+            return E_FAIL;
+        }
+        break;
+    }
```

Comment:
Is this 'break' supposed to be here?

Reply:
break for loop, but loop is not necessary anymore.  Fix: remove while loop